Chore: Add doctrine/orm 3.x and PHPUnit 9+ compatibility

bpolaszek opened the pull request:
Depends on:
- https://github.com/matthiasnoback/doctrine-dbal-test-service-provider/pull/6
- https://github.com/matthiasnoback/phpunit-test-service-container/pull/9

bpolaszek committed:
chore: update library
- Migrate to PHP 8 attributes in Doctrine annotations
- Replace deprecated `Setup` with `ORMSetup`
- Update PHPUnit configuration to newer XML schema
- Add type hints and return types to test methods
- Refactor test method annotations to `test_` prefix

bpolaszek committed:
refactor: implement EntityManager initialization regarding latest recommendations / deprecations

bpolaszek committed:
chore: expand PHPUnit version support to include ^10.0, ^11.0, and ^12.0

bpolaszek committed:
chore: expand Doctrine ORM version support to include ^3.0

matthiasnoback commented:
Hi! Thanks for your PRs. I am not using these projects myself but if you do and want to keep them running, it's okay if you take over maintenance. We'd have to move these projects to a different repository/organization.
If you just want to get these PRs merged, please make the CI work again (Travis doesn't work anymore, so you'd have to add support for GitHub Actions).

bpolaszek committed:
chore: migrate CI to GitHub Actions and update project dependencies

- Replace Travis CI with GitHub Actions for continuous integration
- Update supported PHP versions to ^7.1 and ^8.x
- Adjust composer.json homepage to HTTPS
- Update README build badge to reflect new CI status

bpolaszek commented:
Hi @matthiasnoback,

Not eager to take over maintenance but I just added CI using GA on all libraries. For unknown reasons CI won't be triggered on PRs, but on my own forks.
This one fails because it waits for that one to be merged & tagged:
- https://github.com/matthiasnoback/phpunit-test-service-container/pull/9

https://github.com/bpolaszek/doctrine-orm-test-service-provider/actions

matthiasnoback commented:
Thanks for this one too!
Currently, the build fails. Could you take a look?

bpolaszek commented:
Indeed - I'll have a look.
Can you please merge and tag this one beforehand? 
- https://github.com/matthiasnoback/doctrine-dbal-test-service-provider/pull/6

Tests successfully ran [there](https://github.com/bpolaszek/doctrine-dbal-test-service-provider/actions/runs/19015944091)
Thank you,
Ben

misaert commented:
Hi,

@matthiasnoback is it possible to create the release with the new changes?

Thanks,

Mickaël

matthiasnoback commented:
Sorry, I wasn't aware. Done now!

misaert commented:
Thanks @matthiasnoback :pray:
